NET_project/Benchmarks/Seahash.cs

```diff
@@ -1,58 +1,63 @@
 ﻿using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 namespace NET_project.Benchmarks;
 
 public unsafe struct SeahashNET : IJob {
 	public uint iterations;
 	public ulong result;
 
-	public void Run()
-	{
+	public void Run() {
 		result = Seahash(iterations);
 	}
 
-	private ulong Seahash(uint Iterations)
-	{
+	private static ulong Seahash(uint Iterations) {
 		const int bufferLength = 1024 * 128;
 
-		byte[] buffer = new byte[bufferLength];
+		// Using span here would be more complex due to it unsafely reading
+		// 8 byte pieces of data and depending on alignment to make overreading
+		// it does "safe".
+		byte* buffer = (byte*)NativeMemory.AlignedAlloc(bufferLength, 8);
 
-		for (int i = 0; i < bufferLength; i++)
-		{
+		for (int i = 0; i < bufferLength; i++) {
 			buffer[i] = (byte)(i % 256);
 		}
 
 		ulong hash = 0;
 
-		for (uint i = 0; i < Iterations; i++)
-		{
+		for (uint i = 0; i < Iterations; i++) {
 			hash = Compute(buffer, bufferLength, 0x16F11FE89B0D677C, 0xB480A793D8E6C86C, 0x6FE2E5AAF078EBC9, 0x14F994A4C5259381);
 		}
 
+		NativeMemory.AlignedFree(buffer);
+
 		return hash;
 	}
 
 	[MethodImpl(MethodImplOptions.AggressiveInlining)]
-	private ulong Diffuse(ulong value)
-	{
+	private static ulong Read(byte* pointer) {
+		return *(ulong*)pointer;
+	}
+
+	[MethodImpl(MethodImplOptions.AggressiveInlining)]
+	private static ulong Diffuse(ulong value) {
 		value *= 0x6EED0E9DA4D94A4F;
-		value ^= ((value >> 32) >> (int)(value >> 60));
+		value ^= value >> 32 >> (int)(value >> 60);
 		value *= 0x6EED0E9DA4D94A4F;
 
 		return value;
 	}
 
-	private ulong Compute(byte[] buffer, ulong length, ulong a, ulong b, ulong c, ulong d)
-	{
+	private static ulong Compute(byte* buffer, ulong length, ulong a, ulong b, ulong c, ulong d) {
 		const uint blockSize = 32;
 
 		ulong end = length & ~(blockSize - 1);
 
 		for (uint i = 0; i < end; i += blockSize) {
-			a ^= buffer[i];
-			b ^= buffer[i + 8];
-			c ^= buffer[i + 16];
-			d ^= buffer[i + 24];
+			a ^= Read(buffer + i);
+			b ^= Read(buffer + i + 8);
+			c ^= Read(buffer + i + 16);
+			d ^= Read(buffer + i + 24);
```

Comment:
There was actually a bug here in the previous implementation. It was only processing a `byte`, not a `uint64` as native was doing.